Revert "split up tests"

dralley opened the pull request:
This reverts commit 7f19149bf1a98e47a31b785d2a292b1cc9ab7095.

drahnr commented:
Precisely what I did not want. Please delete those from the rebase that became irrelevant.

dralley commented:
"delete those from the rebase that became irrelevant"?

drahnr commented:
> "delete those from the rebase that became irrelevant"?

The rebased branch of your test changes named `bernhard-foo`.

dralley commented:
I left in `tests.rs` only the tests which could not be done in `tests/...`, the ones using internal-only API that we have no substitute for.  Others which made sense as unit tests I moved to their respective files.  Everything else which used the public-facing API and fixture files I moved to `tests/...`

I didn't delete anything outright or replaced with a test that does the same things.

drahnr commented:
Could you elaborate why they all have to be integration tests?

dralley commented:
They're not *all* functional tests, e.g. the few which I moved to unit tests in header.rs which is also cherrypicked to this PR.  It makes sense to have these but there's no reason for them to be in `src/tests/` rather than in `header.rs` if they're just testing internal implementation details of the header specifically.

But most of them *are* integration / functional tests, not unit tests.  Parsing an RPM and then exercising accessors on it, or parsing an RPM and verifying the signature, or creating an RPM and then signing it and then verifying it - those are functional / integration tests exercising entire workflows, not isolated details.

dralley commented on src/rpm/headers/header.rs:
```diff
@@ -973,3 +915,83 @@ impl IndexData {
+        );
+        assert_eq!(-48, entry.offset);
+
+        Ok(())
```
These are unit tests - and remain as such in my PR.  I just move them into the specific file they apply to.

dralley commented:
I'm just going to close this and rebase my PR on top of it.  The main thing is, I just wanted to keep the meanings of the commits intact and make the reversal explicit.
